### PR TITLE
improve table structure

### PR DESCRIPTION
**Pull request by ltclm**
in order to have huge numbers of features per table we have to change…the pk type from integer to bigint

shapefilepath attribute contains the path of the imported shape file

### CODE REVIEW

**Comment by ltclm** [comment]
This is the modified configuration we are using for the import of the missing shape files sectors NO and NW.

File: forge/db.py [hunk 1]
```diff
@@ -105,8 +105,11 @@ def populateFeatures(args):
         bulk = BulkInsert(model, session, withAutoCommit=1000)
         for feature in shp.getFeatures():
             polygon = feature.GetGeometryRef()
+            # add shapefile path to dict
+            # self.shpFilePath
             bulk.add(dict(
                 the_geom=WKTElement(polygon.ExportToWkt(), 4326)
+                , shapefilepath=shpFile
```
Comment (gjn) [inline forge/db.py]: Just to be sure: this doesn't inrease the db too much? (I know I proposed it)

**Comment by ltclm** [comment]
the table will be bigger, 

```sql
-- id (bigint): 8bytes (size without index)
select pg_column_size(8::bigint); 

-- shapefilepath (text): 28 bytes
select pg_column_size('/geodata/tmp/1175-43.shp'::text); 

-- the_geom (geometry): 144 bytes (size without index)
select pg_column_size(ST_GeomFromEWKT('SRID=4326;POLYGON ((9.456798491739281 46.894738413824633 2138.90130805131048,9.456866951866841 46.894813692054512 2142.730545211583376,9.456837339517289 46.894724231298014 2139.28147372789681,9.456798491739281 46.894738413824633 2138.90130805131048))')::geometry) 

```

Adding the shapefilpath will make a row become about 15% bigger. This is without geometry and id index, so probably around 10%.

Given an estimated total size of 2.5 TB of all data in postgis it will be 2.75 TB.

**Comment by ltclm** [comment]
this pr is now representing the configuration used for the current database release.
the database is ready, will take a snapshot of the ebs tomorrow morning and then we can start with the tile generation.

**Comment by gjn** [comment]
Could you please fix pep.

`forge/db.py:111:65: E203 whitespace before ','
forge/models/tables.py:8:1: F401 'Integer' imported but unused
forge/models/tables.py:31:47: E231 missing whitespace after ','
forge/models/terrain_metadata.py:57:15: E121 continuation line under-indented for hanging indent`

File: tms.cfg.mako [hunk 2]
```diff
@@ -18,8 +18,11 @@ maxLat: 46.898
 fullonly: 0
 
 [Zooms]
-tileMinZ: 9
-tileMaxZ: 17
+tileMinZ: 8
+tileMaxZ: 18
```
Comment (gjn) [inline tms.cfg.mako]: We don't do level 18 for now.

File: tms.cfg.mako [hunk 3]
```diff
@@ -43,10 +46,10 @@ tablename: bl_8m
 tablename: bl_4m
 
 [16]
-tablename: bl_4m
+tablename: bl_2m
 
 [17]
-tablename: bl_4m
+tablename: bl_1m
```
Comment (gjn) [inline tms.cfg.mako]: level 17 should be 0.5, no?

**Comment by ltclm** [comment]
thanks for the feedback, changes have been applied.

**Comment by gjn** [comment]
extent should be updated as well:
`5.86725126512748 45.8026860136571,10.9209100671547 47.8661652478939`

**Comment by gjn** [comment]
Because of layers.json, we need to test-run on a fully existing region. I'll define one around Thun.

**Pull request merged**